src/box/lua/schema.lua

```diff
@@ -1967,12 +1967,18 @@ local function normalize_position(index, pos, ret_pos, ret_pos_end)
         ret_pos_end[0] = ret_pos[0] + #pos
     else
         local tuple_ibuf = cord_ibuf_take()
-        local tuple, tuple_end = tuple_encode(tuple_ibuf, pos)
-        local nok = builtin.box_index_tuple_position(index.space_id, index.id,
-                                                     tuple, tuple_end,
-                                                     ret_pos, ret_pos_end) ~= 0
+        local ok, res = pcall(function()
+            local tuple, tuple_end = tuple_encode(tuple_ibuf, pos)
+            return builtin.box_index_tuple_position(index.space_id, index.id,
+                                                    tuple, tuple_end,
+                                                    ret_pos, ret_pos_end) == 0
+        end)
```

Comment:
In the [original discussion](https://lists.tarantool.org/tarantool-patches/06d4c252-3f41-d91b-6943-ec4cca520a99@tarantool.org/) (which is mentioned in a [comment](https://github.com/tarantool/tarantool/issues/8677#issuecomment-1556740005) to this issue), @Gerold103 claimed that not using `pcall` was an intentional design decision to avoid possible performance degradation. Have you run any performance tests to disprove his statement?

Reply:
<details>
<summary>I did a simple benchmark:</summary>

```lua
os.execute('rm -rf *.snap *.xlog *.vylog 512')

clock = require('clock')
log = require('log')

box.cfg{}

s = box.schema.space.create('s')
s:create_index('pk')
s:insert{0}

N = 100
total_time = 0
for _ = 1, N do
    total_time = total_time + clock.bench(function() box.space.s:get{0} end)[1]
end

log.info("Average time: %g ms", total_time / N * 1e6)

os.exit()
```
</details>

Results:

| Branch         | Time, ms        |
| -------------- | --------------- |
| Master         | 4.79248         |
| Patch          | 5.18757         |